packages/cli/src/commands/generate/auth/auth.js

```diff
@@ -80,30 +124,42 @@ export const files = (provider) => {
 }
 
 // actually inserts the required config lines into index.js
-export const addConfigToIndex = (config) => {
+export const addConfigToIndex = async (config, force) => {
   let content = fs.readFileSync(WEB_SRC_INDEX_PATH).toString()
 
-  content = addWebImports(content, config.imports)
-  content = addWebInit(content, config.init)
-  content = addWebRender(content, config.authProvider)
+  // update existing AuthProvider if --force else add new AuthProvider
+  if (content.includes(AUTH_PROVIDER_IMPORT) && force) {
+    content = await removeOldAuthProvider(content)
+    content = addWebInit(content, config.init)
+    content = updateWebRender(content, config.authProvider)
+  } else {
+    content = addWebImports(content, config.imports)
+    content = addWebInit(content, config.init)
+    content = addWebRender(content, config.authProvider)
+  }
 
   fs.writeFileSync(WEB_SRC_INDEX_PATH, content)
 }
 
 export const addApiConfig = () => {
   let content = fs.readFileSync(API_GRAPHQL_PATH).toString()
-
-  // add import statement
-  content = content.replace(
-    /^(.*services.*)$/m,
-    `$1\n\nimport { getCurrentUser } from 'src/lib/auth'`
+  const [_, hasAuthImport] = content.match(
+    /(import {.*} from 'src\/lib\/auth.*')/s
```

Comment:
This is throwing an error:
```bash
object null is not iterable (cannot read property Symbol(Symbol.iterator))
```

But I manually checked logic for case true | false and it's working well.

Reply:
for `.match()` "Matches a string or an object that supports being matched against, and returns an array containing the results of that search, or null if no matches are found."

So I believe what's happening in the case where project does **not** already have auth, it returns `null` and throws.